### ISSUE TITLE
Base64 Conversion Not Working?

### ISSUE THREAD

**Issue by ForeverZer0** [ohler55/ox#213]
From the the documentation:

> Strings with characters not allowed in XML are base64 encoded amd will be converted back into a String when loaded.

Running a simple test:
```Ruby
class Illegal

  def initialize
    @bad_char_1 = "\v"
    @bad_char_2 = "\f"
  end
end

test = Illegal.new
Ox.dump(test)
```

Gives this result:

> Exception `SyntaxError' at -e:1 - '\#x0c' is not a valid XML character.

Running Ruby 2.4.4 (32-bit) on Windows 10 (64-bit)

Am I missing an option, or is this not the expected behavior?

EDIT:
In addition, a null character "\0", which is also "illegal" does not throw an error, but is ignored. There is no conversion, the node is simply empty.

**Comment by ohler55** [comment]
Encoding into base64 was removed although it still exists for legacy XML loading. Instead, if the `:invalid_replace` option is used to deal with invalid characters. 

As for `'\0'` that should fall into the same category. I'll look into why that is not working as expected

**Comment by ohler55** [comment]
Sorry it took a while for me to get back to you. I ran the test on Mac and it worked. A `"\0abc"` ended up as an empty string so that is not correct. Is there something else that is an issue with the current version of Ox?

**Comment by ohler55** [comment]
Fixed in release 2.9.3 which was just released. Thanks.

**Comment by ForeverZer0** [comment]
Not more issues that I have come across, seems to be rather robust in its current state.
No problem on the wait, wasn't holding me back from anything.

Thanks again. 👍
